main.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"encoding/csv"
+	"flag"
+	"io"
+	"log"
+	"os"
+)
+
+func main() {
+	//TODO move args parsing to a function
+	configFile := flag.String("config", "config.json", "Configuration of the data to be anonymised. Default is 'config.json'")
+	outputFile := flag.String("output", "", "Output file. Default is stdout.")
+	flag.Parse()
+	log.Printf("Using configuration in file %s\n", *configFile)
+	r := initReader(flag.Arg(0))
+	w := initWriter(*outputFile)
+	//TODO create the anons array using the config provided
+	anons := []anonymisation{hash, identity, identity, outcode}
+	i := 0
+
+	for {
+		record, err := r.Read()
```

Comment:
Not knowing how Go works, is this call buffered properly? Just otherwise I'm worried this may actually result in a read call per line which would perform awfully. In JVM land you'd obviously use a `BufferedReader` of some kind.

Reply:
Yep,  it uses a `bufio`: https://github.com/golang/go/issues/16791.

Reply:
My guess was that Go does it properly but I was also hoping for @Briansbum to take a look at these things.